app/models/comment.rb

```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  belongs_to :post
+
+  def update_comments_counter
+    post.update(comments_counter: post.comments.count)
+  end
```

Comment:
- [ Optional ] Kindly I recommend you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_comments_counterr method every time a new post has been saved to database, then make the update_comments_counter method private Just Like the below suggestion  .

 ```
after_save :update_comments_counter

 private

   def update_comments_counter
    post.update(comments_counter: post.comments.count)
  end

```